### PR TITLE
fix: only return visible posts

### PR DESCRIPTION
**Pull request by rebelchris**
We should not return non visible posts on id/companion endpoints.

AS-629 #done

### CODE REVIEW

**Review by capJavert — APPROVED** [review]
If possible test case 🚀

File: src/schema/posts.ts [hunk 1]
```diff
@@ -1109,7 +1109,7 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
       const partialPost = await ctx.con.getRepository(Post).findOneOrFail({
         select: ['id', 'sourceId', 'private'],
         relations: ['source'],
-        where: [{ id }, { slug: id }],
+        where: [{ id }, { slug: id }, { visible: true }],
```
[the remark this answers is not in the document]
Comment (rebelchris) [reply src/schema/posts.ts]: Not sure on product need, think we allow it to the owner still? (or if bookmarked etc?) (not 100% sure but feels there might be a edge-case)

**Comment by rebelchris** [comment]
@capJavert Thanks! I actually missed the where clause and made it accidentally `or or or` 😅
New tests should work.

File: src/schema/posts.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (capJavert) [reply src/schema/posts.ts]: In graphorm we actually do:
```
qb
  .andWhere(`"${alias}"."deleted" = false`)
  .andWhere(`"${alias}"."visible" = true`)
```

so maybe deleted actually instead of banned

**Review by omBratteng — APPROVED** [review]
lgtm

File: src/schema/posts.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (rebelchris) [reply src/schema/posts.ts]: @capJavert Done, actually cleaned it up so we better re-use same evaluations.

**Review by capJavert — APPROVED** [review]
